search-api/search_api/resources/businesses.py

```diff
@@ -0,0 +1,144 @@
+# Copyright © 2020 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+from tempfile import NamedTemporaryFile
+
+from flask import Blueprint, request, jsonify, send_from_directory
+from openpyxl import Workbook
+
+from search_api.auth import jwt, authorized
+from search_api.models import (
+    Corporation,
+    CorpName,
+    Office,
+    _get_corporation_search_results,
+    _merge_corpparty_search_addr_fields,
+    _normalize_addr,
+    _format_office_typ_cd
+)
+
+
+API = Blueprint('BUSINESSES_API', __name__, url_prefix='/api/v1/businesses')
+
+
+@API.route('/search/')
+@jwt.requires_auth
+def corporation_search():
+    # TODO SY - check roles
+    # check authorization
```

Comment:
Check authorization here, there should be a call to auth-api to see if the user has access to that account and Director Search.

Reply:
Do you know how we should be checking for access to an account and to Director Search? are there roles we can use?

Reply:
GET` https://auth-api-dev.pathfinder.gov.bc.ca/api/v1/accounts/<account_id>/products/DIR_SEARCH/authorizations` 

account_id : The current account user is in;  You can get it from the CURRENT_ACCOUNT value in session storage. 
If the response have the value 'orgMembership', then you can allow. Else don't allow.

e.g;
```
{
  "orgMembership": "OWNER", 
  "roles": []
}
```

Reply:
An extra auth API call for every search API call seems likely to cause a performance hit.

Is there a way to have the auth API include these memberships as claims in the JWT, to avoid another API call? I thought the purpose of JWT was to use the signed token to indicate the user's auth scope without needing to call an external service.

Reply:
The token is signed during authentication and doesn't have the account context. An user can be part of more than one account, and can have different roles/permissions on different account. So the token provides high level role to access the application and endpoints; For fine grained authorization it needs to call auth-api (authorization). 
All the services (pay, legal etc.) does the same. The /authorizations endpoint is developed in a way to have high performance and so far we haven't seen any performance issues in any systems.